### PR TITLE
Fix a bug when alias name and its first value overlap

### LINKED ISSUE #141: The completion isn't shown when an alias ends with whitespaces
### OS version

```shell
macOS Big Sur
```


### zsh version

```shell
zsh 5.8 (x86_64-apple-darwin18.7.0)
```


### awk version

```shell
awk version 20200816
```


### jq version

```shell
jq-1.6
```


### fzf version

```shell
0.27.2 (e086f0b)
```


### Command version(s)

```shell
None
```


### Locale

```shell
LANG="ja_JP.UTF-8"
LC_COLLATE="ja_JP.UTF-8"
LC_CTYPE="ja_JP.UTF-8"
LC_MESSAGES="ja_JP.UTF-8"
LC_MONETARY="ja_JP.UTF-8"
LC_NUMERIC="ja_JP.UTF-8"
LC_TIME="ja_JP.UTF-8"
LC_ALL=
```


### Installed zsh plugins (if applicable)

```shell
```


### What happened?

When an alias ends with whitespaces, the completion isn't shown and a large quantity of whitespaces appear after the aliased command as below:

```sh
fzf-zsh-completions +(master)  $  sudo                                                                                    
                                                                                                                          
                                                                                                                          
                                              **
```

The number of whitespaces is 374 but may not be correct. For your reference, `$COLUMNS` of My terminal is 122. Also, an alias ending with whitespaces is valid as the citing below:

> If the replacement text ends with a space, the next word in the shell input is always eligible for purposes of alias expansion.
>
> <cite>zshmisc(1)</cite>

### What did you expect to happen?

The completion is shown, and many whitespaces disappear.

### Steps to reproduce

```sh
fzf-zsh-completions +(master)  $  alias sudo='sudo '
fzf-zsh-completions +(master)  $  source fzf-zsh-completions.plugin.zsh
fzf-zsh-completions +(master)  $  sudo **<Tab>
```

### PR DESCRIPTION
**Pull request by hexium310**
Resolves #141

### CODE REVIEW

File: src/core/aliases.zsh [hunk 1]
```diff
@@ -12,6 +12,7 @@ _fzf_complete_enable_aliases() {
         if [[ -n $completer ]]; then
             eval "
                 _fzf_complete_$name() {
+                    source ${@[(r)*completers/$completer.zsh]}
```
Comment (chitoku-k) [inline src/core/aliases.zsh:15]: I don't like this function sources every time this complete function is called. How about just embedding the content of that function?

Comment (hexium310) [reply src/core/aliases.zsh:15]: Does it mean to insert `$(declare -f _fzf_complete_$completer)`?

Comment (chitoku-k) [reply src/core/aliases.zsh:15]: No. `$(cat -- $[@(r)*completers/$completer.zsh]})`

**Comment by hexium310** [comment]
Using the same name as the original for the name of the wrapper function overwrites the original function after running once😢

**Comment by hexium310** [comment]
It seems that this is correctly running, and there isn't a strange function declaration. What do you think, @chitoku-k?

**Comment by chitoku-k** [comment]
We can retrieve a function body from `$functions[function_name]`.

**Comment by chitoku-k** [comment]
How about this?
```
eval "
    _fzf_complete_$name() {
        LBUFFER=\"\${LBUFFER/$name/$completer $arguments}\"
        () {
            $(cat -- $[@(r)*completers/$completer.zsh]})
        } \${@/$name/$completer $arguments}
        LBUFFER=\"\${LBUFFER/$completer $arguments/$name}\"
    }
"
```

**Comment by hexium310** [comment]
Simple and good looking!

**Comment by chitoku-k** [comment]
Please update this PR in the way I described above and I'll merge this.

**Review by chitoku-k — APPROVED** [review]
Thanks!

**Pull request merged**